example_cpp_logistic_regression fail on 4.11.0 version

### System Information

OpenCV version: 4.11.0 Operating System / Platform: Ubuntu 22.04 Compiler & compiler version: GCC 14.2.0

### Detailed description

```
============== example_cpp_logistic_regression start ================
[ WARN:0@0.002] global samples.cpp:61 findFile cv::samples::findFile('data01.xml') => '/usr/share/OpenCV/samples/data/data01.xml'
**********************************************************************
/usr/share/OpenCV/samples/data/data01.xml contains digits 0 and 1 of 20 samples each, collected on an Android device
Each of the collected images are of size 28 x 28 re-arranged to 1 x 784 matrix
**********************************************************************
loading the dataset...read 40 rows of data
training/testing samples count: 20/20
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.11.0) /usr/src/debug/opencv/4.11.0.imx/modules/ml/src/lr.cpp:129: error: (-5:Bad argument) data should have at least 2 classes in function 'train'

/mnt/nfs/L6.12.3_1.0.0_Q1_RC1/vte_mx95/testcases/bin/gpu_opencv_test.sh: line 526: 19683 Aborted                 (core dumped) example_cpp_logistic_regression
wait 15 secs ...
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
TFAIL: The app example_cpp_logistic_regression is not killed successfully in 60 seconds, will exit test.
Following apps failed, please check:
example_cpp_logistic_regression
```

### Steps to reproduce

./example_cpp_logistic_regression

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/samples/data/data01.xml
+++ b/samples/data/data01.xml
@@ -50,7 +50,7 @@
 <labelsmat type_id="opencv-matrix">
     <rows>40</rows>
     <cols>1</cols>
-    <dt>d</dt>
+    <dt>f</dt>
     <data>
         0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
     </data>

